[libmad] Build error on wasm32-emscripten

Package: libmad:wasm32-emscripten@0.16.4#1

**Host Environment**

- Host: arm64-osx
- Compiler: Clang 19.0.0
-    vcpkg-tool version: 2024-01-11-710a3116bbd615864eef5f9010af178034cb9b44
    vcpkg-scripts version: 215a25355 2024-03-01 (3 hours ago)

**To Reproduce**

`vcpkg install libmad`

**Failure logs**

```
-- Downloading https://codeberg.org/tenacityteam/libmad/releases/download/0.16.4/libmad-0.16.4.tar.gz -> tenacityteam-libmad-0.16.4.tar.gz...
-- Extracting source /Users/<user>/git/vcpkg/downloads/tenacityteam-libmad-0.16.4.tar.gz
-- Using source at /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean
-- Found external ninja('1.11.1').
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /Applications/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j11
    Working Directory: /Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg
    See logs for more information:
      /Users/<user>/git/vcpkg/buildtrees/libmad/install-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/libmad/portfile.cmake:20 (vcpkg_cmake_install)
  scripts/ports.cmake:172 (include)



```
<details><summary>.../vcpkg/buildtrees/libmad/install-wasm32-emscripten-dbg-out.log</summary>

```
Change Dir: '/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg'

Run Build Command(s): /opt/homebrew/bin/ninja -v -v -j11 install
[1/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/version.c.o -MF CMakeFiles/mad.dir/version.c.o.d -o CMakeFiles/mad.dir/version.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/version.c
[2/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/fixed.c.o -MF CMakeFiles/mad.dir/fixed.c.o.d -o CMakeFiles/mad.dir/fixed.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/fixed.c
[3/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/stream.c.o -MF CMakeFiles/mad.dir/stream.c.o.d -o CMakeFiles/mad.dir/stream.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/stream.c
[4/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/bit.c.o -MF CMakeFiles/mad.dir/bit.c.o.d -o CMakeFiles/mad.dir/bit.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/bit.c
[5/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/layer12.c.o -MF CMakeFiles/mad.dir/layer12.c.o.d -o CMakeFiles/mad.dir/layer12.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c
FAILED: CMakeFiles/mad.dir/layer12.c.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/layer12.c.o -MF CMakeFiles/mad.dir/layer12.c.o.d -o CMakeFiles/mad.dir/layer12.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:95:10: error: invalid output constraint '=a' in asm
   95 |   return mad_f_mul(sample, linear_table[nb - 2]);
      |          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:95:10: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:213:4: error: invalid output constraint '=a' in asm
  213 |           mad_f_mul(I_sample(&stream->ptr, nb, stream),
      |           ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:213:4: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:235:6: error: invalid output constraint '=a' in asm
  235 |             mad_f_mul(sample, sf_table[scalefactor[ch][sb]]);
      |             ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:235:6: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:368:17: error: invalid output constraint '=a' in asm
  368 |     output[s] = mad_f_mul(requantized + quantclass->D, quantclass->C);
      |                 ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:368:17: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:579:8: error: invalid output constraint '=a' in asm
  579 |               mad_f_mul(samples[s], sf_table[scalefactor[ch][sb][gr / 4]]);
      |               ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:579:8: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:600:8: error: invalid output constraint '=a' in asm
  600 |               mad_f_mul(samples[s], sf_table[scalefactor[ch][sb][gr / 4]]);
      |               ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer12.c:600:8: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
12 errors generated.
[6/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/decoder.c.o -MF CMakeFiles/mad.dir/decoder.c.o.d -o CMakeFiles/mad.dir/decoder.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/decoder.c
[7/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/frame.c.o -MF CMakeFiles/mad.dir/frame.c.o.d -o CMakeFiles/mad.dir/frame.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/frame.c
[8/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/synth.c.o -MF CMakeFiles/mad.dir/synth.c.o.d -o CMakeFiles/mad.dir/synth.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c
FAILED: CMakeFiles/mad.dir/synth.c.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/synth.c.o -MF CMakeFiles/mad.dir/synth.c.o.d -o CMakeFiles/mad.dir/synth.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:212:35: error: invalid output constraint '=a' in asm
  212 |   t0   = in[0]  + in[31];  t16  = MUL(in[0]  - in[31], costab1);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:212:35: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:213:35: error: invalid output constraint '=a' in asm
  213 |   t1   = in[15] + in[16];  t17  = MUL(in[15] - in[16], costab31);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:213:35: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:216:10: error: invalid output constraint '=a' in asm
  216 |   t59  = MUL(t16 - t17, costab2);
      |          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:216:10: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:218:10: error: invalid output constraint '=a' in asm
  218 |   t50  = MUL(t0  - t1,  costab2);
      |          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:218:10: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:220:35: error: invalid output constraint '=a' in asm
  220 |   t2   = in[7]  + in[24];  t18  = MUL(in[7]  - in[24], costab15);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:220:35: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:221:35: error: invalid output constraint '=a' in asm
  221 |   t3   = in[8]  + in[23];  t19  = MUL(in[8]  - in[23], costab17);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:221:35: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:224:10: error: invalid output constraint '=a' in asm
  224 |   t60  = MUL(t18 - t19, costab30);
      |          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:224:10: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:226:10: error: invalid output constraint '=a' in asm
  226 |   t51  = MUL(t2  - t3,  costab30);
      |          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:226:10: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:228:35: error: invalid output constraint '=a' in asm
  228 |   t4   = in[3]  + in[28];  t20  = MUL(in[3]  - in[28], costab7);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:228:35: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:229:35: error: invalid output constraint '=a' in asm
  229 |   t5   = in[12] + in[19];  t21  = MUL(in[12] - in[19], costab25);
      |                                   ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/synth.c:109:22: note: expanded from macro 'MUL'
  109 | #  define MUL(x, y)  mad_f_mul((x), (y))
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[9/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/huffman.c.o -MF CMakeFiles/mad.dir/huffman.c.o.d -o CMakeFiles/mad.dir/huffman.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/huffman.c
[10/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/timer.c.o -MF CMakeFiles/mad.dir/timer.c.o.d -o CMakeFiles/mad.dir/timer.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/timer.c
[11/13] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/layer3.c.o -MF CMakeFiles/mad.dir/layer3.c.o.d -o CMakeFiles/mad.dir/layer3.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c
FAILED: CMakeFiles/mad.dir/layer3.c.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/emcc -DASO_ZEROCHECK -DHAVE_ASSERT_H -DHAVE_FCNTL_H -DHAVE_FORK -DHAVE_FTRUNCATE -DHAVE_LIMITS_H -DHAVE_PIPE -DHAVE_SYS_MMAN_H -DHAVE_SYS_STAT_H -DHAVE_SYS_TYPES_H -DHAVE_SYS_WAIT_H -DHAVE_UNISTD_H -DHAVE_WAITPID -DOPT_ACCURACY -I/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean -I/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg -g -MD -MT CMakeFiles/mad.dir/layer3.c.o -MF CMakeFiles/mad.dir/layer3.c.o.d -o CMakeFiles/mad.dir/layer3.c.o -c /Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:947:17: error: invalid output constraint '=a' in asm
  947 |   return frac ? mad_f_mul(requantized, root_table[3 + frac]) : requantized;
      |                 ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:947:17: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1524:17: error: invalid output constraint '=a' in asm
 1524 |             opposite = mad_f_mul(left, lsf_scale[(is_pos - 1) / 2]);
      |                        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1524:17: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1555:19: error: invalid output constraint '=a' in asm
 1555 |           xr[0][l + i] = mad_f_mul(left, is_table[    is_pos]);
      |                          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1555:19: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1556:19: error: invalid output constraint '=a' in asm
 1556 |           xr[1][l + i] = mad_f_mul(left, is_table[6 - is_pos]);
      |                          ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1556:19: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1583:17: error: invalid output constraint '=a' in asm
 1583 |         xr[0][l + i] = mad_f_mul(m + s, invsqrt2);  /* l = (m + s) / sqrt(2) */
      |                        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1583:17: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1584:17: error: invalid output constraint '=a' in asm
 1584 |         xr[1][l + i] = mad_f_mul(m - s, invsqrt2);  /* r = (m - s) / sqrt(2) */
      |                        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1584:17: error: invalid input constraint 'I' in asm
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:493:8: note: expanded from macro 'mad_f_mul'
  493 |        mad_f_scale64(__hi, __lo);  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1615:2: error: invalid output constraint '=a' in asm
 1615 |         MAD_F_ML0(hi, lo,  a, cs[i]);
      |         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:505:35: note: expanded from macro 'MAD_F_ML0'
  505 | #  define MAD_F_ML0(hi, lo, x, y)       MAD_F_MLX((hi), (lo), (x), (y))
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1616:2: error: invalid output constraint '=a' in asm
 1616 |         MAD_F_MLA(hi, lo, -b, ca[i]);
      |         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:224:8: note: expanded from macro 'MAD_F_MLA'
  224 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1618:15: error: invalid input constraint 'I' in asm
 1618 |         xr[-1 - i] = MAD_F_MLZ(hi, lo);
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:513:30: note: expanded from macro 'MAD_F_MLZ'
  513 | #  define MAD_F_MLZ(hi, lo)             mad_f_scale64((hi), (lo))
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1620:2: error: invalid output constraint '=a' in asm
 1620 |         MAD_F_ML0(hi, lo,  b, cs[i]);
      |         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:505:35: note: expanded from macro 'MAD_F_ML0'
  505 | #  define MAD_F_ML0(hi, lo, x, y)       MAD_F_MLX((hi), (lo), (x), (y))
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1621:2: error: invalid output constraint '=a' in asm
 1621 |         MAD_F_MLA(hi, lo,  a, ca[i]);
      |         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:224:8: note: expanded from macro 'MAD_F_MLA'
  224 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1623:15: error: invalid input constraint 'I' in asm
 1623 |         xr[     i] = MAD_F_MLZ(hi, lo);
      |                      ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:513:30: note: expanded from macro 'MAD_F_MLZ'
  513 | #  define MAD_F_MLZ(hi, lo)             mad_f_scale64((hi), (lo))
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:249:34: note: expanded from macro 'mad_f_scale64'
  249 |             : "0" (__lo_), "r" (__hi_), "I" (MAD_F_SCALEBITS)  \
      |                                         ^
/Users/<user>/git/vcpkg/buildtrees/libmad/src/m-libmad-0-b009386def.clean/layer3.c:1672:8: error: invalid output constraint '=a' in asm
 1672 |   m0 = mad_f_mul(a17, -c3);
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:492:8: note: expanded from macro 'mad_f_mul'
  492 |        MAD_F_MLX(__hi, __lo, (x), (y));  \
      |        ^
/Users/<user>/git/vcpkg/buildtrees/libmad/wasm32-emscripten-dbg/mad.h:213:5: note: expanded from macro 'MAD_F_MLX'
  213 |          : "=a" (lo), "=d" (hi)  \
      |            ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/libmad/portfile.cmake
+++ b/ports/libmad/portfile.cmake
@@ -11,9 +11,26 @@ vcpkg_download_distfile(
 
 vcpkg_extract_source_archive(SOURCE_PATH ARCHIVE "${ARCHIVE}")
 
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        aso ASO
+)
+
+set(EXTRA_OPTIONS)
+
+# Avoid architecture-specific assembly when targeting WASM.  The upstream
+# CMakeLists incorrectly recognizes the CPU as an Intel/64-bit CPU, therefore
+# we have to override these flags:
+# https://codeberg.org/tenacityteam/libmad/src/commit/84ba587793d61caadf6d1f6c0d94c3e165874a50/CMakeLists.txt
+if(VCPKG_TARGET_IS_EMSCRIPTEN)
+    list(APPEND EXTRA_OPTIONS "-DFPM_64BIT=OFF -DFPM_INTEL=OFF -DFPM_DEFAULT=ON")
+endif()
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        ${FEATURE_OPTIONS}
+        ${EXTRA_OPTIONS}
         -DEXAMPLE=OFF
 )
 

--- a/ports/libmad/vcpkg.json
+++ b/ports/libmad/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libmad",
   "version": "0.16.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "high-quality MPEG audio decoder",
   "homepage": "http://codeberg.org/tenacityteam/libmad",
   "dependencies": [
@@ -13,5 +13,11 @@
       "name": "vcpkg-cmake-config",
       "host": true
     }
-  ]
+  ],
+  "features": {
+    "aso": {
+      "description": "Enable CPU architecture-specific optimizations (x86, ARM and MIPS only)",
+      "supports": "x86 | x64 | arm"
+    }
+  }
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4562,7 +4562,7 @@
     },
     "libmad": {
       "baseline": "0.16.4",
-      "port-version": 1
+      "port-version": 2
     },
     "libmagic": {
       "baseline": "5.45",

--- a/versions/l-/libmad.json
+++ b/versions/l-/libmad.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "204aa1729bb363c40a79cbcdb3beab84fa4bd03e",
+      "version": "0.16.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "7016c2c6c5e65a6420a176aa97696897bb134c40",
       "version": "0.16.4",
       "port-version": 1

